src/App.tsx

```diff
@@ -1,24 +1,311 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useEffect,
+  useState,
+  ChangeEvent,
+  FormEvent,
+} from 'react';
 
-const USER_ID = 0;
+import { Header } from './components/Header';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
+import { ErrorPanel } from './components/ErrorPanel';
+
+import { Todo } from './types/Todo';
+import { Filters } from './types/Filters';
+import { Errors } from './types/Errors';
+
+import {
+  getTodos,
+  addTodo,
+  removeTodo,
+  updateTodoStatus,
+  updateTodoTitle,
+} from './api/todos';
+
+const USER_ID = 6160;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<Errors>(Errors.NoError);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todos);
+  const [selectedFilter, setSelectedFilter] = useState<Filters>(Filters.All);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [isNewTodoLoading, setIsNewTodoLoading] = useState(false);
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const saveEditedTitle = (
+    todoId: number,
+    todoTitle: string,
+    setEditingTodoId: (value: number) => void,
+  ) => {
+    setLoadingTodoIds([...loadingTodoIds, todoId]);
+    updateTodoTitle(todoId, todoTitle)
+      .then(() => {
+        getTodos(USER_ID)
+          .then(todosFromServer => {
+            setTodos(todosFromServer);
+            setFilteredTodos(todosFromServer);
+            setLoadingTodoIds([]);
+            setEditingTodoId(0);
+          })
+          .catch(() => {
+            setError(Errors.CantUpdate);
+          });
+      });
+  };
+
+  const toggleTodoStatus = (todoId: number, completed: boolean) => {
+    updateTodoStatus(todoId, !completed)
+      .then(() => {
+        setLoadingTodoIds([...loadingTodoIds, todoId]);
+        getTodos(USER_ID)
+          .then(todosFromServer => {
+            setTodos(todosFromServer);
+            setFilteredTodos(todosFromServer);
+            setLoadingTodoIds([]);
+          })
+          .catch(() => {
+            setError(Errors.CantUpdate);
+          });
+      });
+  };
+
+  const getUpdatedTodos = (
+    todosToUpdate: Todo[],
+    everyTodoCompleted: boolean,
+    noTodoCompleted: boolean,
+    someTodosCompleted: boolean,
+  ) => {
+    const updatedTodos: Todo[] = [];
```

Comment:
you can provide an initial value in args area
```suggestion
    todosToUpdate: Todo[] = [],
    everyTodoCompleted: boolean,
    noTodoCompleted: boolean,
    someTodosCompleted: boolean,
  ) => {
```